src/descriptor/bare.rs

```diff
@@ -63,6 +63,26 @@ impl<Pk: MiniscriptKey> Bare<Pk> {
     }
 }
 
+impl<Pk: MiniscriptKey + ToPublicKey> Bare<Pk> {
```

Comment:
I think all this non-failing methods can be re-organized into an additional trait, let's say `PreTaprootDescriptor: Descriptor`, such that all pre-taproot descriptor types may be used as a generalized function arguments

Reply:
We could do for the release. But I think that the effort is not worth it considering everything will go back to non-fallible once we have a new major breaking release of rust-secp that has static contexts